staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -743,3 +743,42 @@ func (s *Scheme) Name() string {
 // internalPackages are packages that ignored when creating a default reflector name. These packages are in the common
 // call chains to NewReflector, so they'd be low entropy names for reflectors
 var internalPackages = []string{"k8s.io/apimachinery/pkg/runtime/scheme.go"}
+
+// ToOpenAPIName returns the REST-friendly OpenAPI name of the groupVersionKind.
+func (s *Scheme) ToOpenAPIName(groupVersionKind schema.GroupVersionKind) (string, error) {
+	example, err := s.New(groupVersionKind)
+	if err != nil {
+		return "", err
+	}
+	rtype := reflect.TypeOf(example).Elem()
+	name := ToRESTFriendlyName(rtype.PkgPath() + "." + rtype.Name())
+	return name, nil
+}
+
+// ToRESTFriendlyName converts Golang package/type canonical name into REST friendly OpenAPI name.
+//
+// Examples of REST friendly OpenAPI name:
+//
+//	Input:  k8s.io/api/core/v1.Pod
+//	Output: io.k8s.api.core.v1.Pod
+//
+//	Input:  k8s.io/api/core/v1
+//	Output: io.k8s.api.core.v1
+//
+//	Input:  csi.storage.k8s.io/v1alpha1.CSINodeInfo
+//	Output: io.k8s.storage.csi.v1alpha1.CSINodeInfo
+//
+// Note that this is a copy of ToRESTFriendlyName from k8s.io/kube-openapi/pkg/util. It is duplicated here to avoid
+// a dependency on kube-openapi.
+func ToRESTFriendlyName(name string) string {
```

Comment:
We currently have 5 copies of these, so collapsing them so they don't drift sounds like a good idea

I would name the exported method more precisely, something like `ToOpenAPIDefinitionName`, and document the input is expected to be `PkgPath + "." TypeName

Reply:
I was able to make this one private.  Many of the other places where we needed this function we already have a dep on kube-openapi so I've updated to call the kube-openapi `ToRESTFriendlyName` directly in those cases and only keep one private copy for runtime.